lib/matplotlib/patheffects.py

```diff
@@ -359,3 +361,164 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
+                                 [np.sin(thetar), np.cos(thetar)]])
+
+        if self._flip:
+            iflip = -1
+        else:
+            iflip = 1
+
+        # Convert spacing parameter to pixels.
+        spcpx = renderer.points_to_pixels(self._spacing)
+
+        # Transform before evaluation because to_polygons works at resolution
+        # of one -- assuming it is working in pixel space.
+        transpath = affine.transform_path(tpath)
+
+        # Evaluate path to straight line segments that can be used to
+        # construct line ticks.
+        polys = transpath.to_polygons(closed_only=False)
+
+        for p in polys:
+            x = p[:, 0]
+            y = p[:, 1]
+
+            # Can not interpolate points or draw line if only one point in
+            # polyline.
+            if x.size < 2:
+                continue
+
+            # Find distance between points on the line
+            ds = np.hypot(x[1:] - x[:-1], + y[1:] - y[:-1])
+
+            # Build parametric coordinate along curve
+            s = np.concatenate(([0.0], np.cumsum(ds)))
+            stot = s[-1]
+
+            num = int(np.ceil(stot / spcpx))
+            # Pick parameter values for ticks.
+            s_tick = np.linspace(0, stot, num)
+
+            # Find points along the parameterized curve
+            x_tick = np.interp(s_tick, s, x)
+            y_tick = np.interp(s_tick, s, y)
+
+            # Find unit vectors in local direction of curve
+            delta_s = self._spacing * .001
+            u = (x_tick - np.interp(s_tick + delta_s, s, x)) / delta_s
+            v = (y_tick - np.interp(s_tick + delta_s, s, y)) / delta_s
+
+            # Normalize slope into unit slope vector.
+            n = np.hypot(u, v)
+            mask = n == 0
+            n[mask] = 1.0
+
+            uv = np.array([u / n, v / n]).T
+            uv[mask] = np.array([0, 0]).T
+
+            # Rotate and scale unit vector into tick vector
+            dxy = np.dot(uv, trans_matrix) * self._length * spcpx
+
+            # Build tick endpoints
+            x_end = x_tick + iflip * dxy[:, 0]
+            y_end = y_tick + iflip * dxy[:, 1]
+
+            # Interleave ticks to form Path vertices
+            xyt = np.empty((2 * num, 2), dtype=x_tick.dtype)
+            xyt[0::2, 0] = x_tick
+            xyt[1::2, 0] = x_end
+            xyt[0::2, 1] = y_tick
+            xyt[1::2, 1] = y_end
+
+            # Transform ticks back to data space before plotting
+            xy = affine.inverted().transform(xyt)
+
+            # Build up vector of Path codes
+            c = np.ones(2 * num, int) * Path.MOVETO
+            c[1::2] = Path.LINETO
+
+            # Construct and draw resulting path
+            h = Path(xy, c)
+            renderer.draw_path(gc0, h, trans, rgbFace)
+
+        gc0.restore()
+
+
+class withTickedStroke(TickedStroke):
+    """
+    Adds a :class:`TickedStroke` and then draws the original
```

Comment:
```suggestion
    Adds a `.TickedStroke` and then draws the original
```
We use sphinx default roles to have cleaner docstrings. https://matplotlib.org/devel/documenting_mpl.html#referring-to-other-code.